### PR TITLE
refactor: set up new directory structure and nx

### PR DESCRIPTION
**Pull request by Lodin**

### CODE REVIEW

File: .eslintrc.json
Comment (tamasmak) [inline .eslintrc.json:1]: The `ignorePatterns` in this file still contains the old directory structure.

File: .gitignore
Comment (tamasmak) [inline .gitignore:1]: The `sso-kit-client/node_modules` line 25 should be also removed.

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitReturns": true,
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "noImplicitOverride": true,
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "experimentalDecorators": true,
+    "useDefineForClassFields": false,
+    "baseUrl": "observability-kit-demo-hilla/frontend",
```
Comment (tamasmak) [inline tsconfig.json:23]: This and the next few lines are not related to the sso-kit.

File: packages/ts/sso-kit-client-core/README.md
Comment (tamasmak) [inline packages/ts/sso-kit-client-core/README.md:1]: The two links in the first line should be fixed according to the new structure.

File: packages/ts/sso-kit-client-core/package.json
Comment (tamasmak) [inline packages/ts/sso-kit-client-core/package.json:1]: The "main" and the "repository:directory" fields should be updated.

File: packages/ts/sso-kit-client-lit/package.json
Comment (tamasmak) [inline packages/ts/sso-kit-client-lit/package.json:1]: The "main" and the "repository:directory" fields should be updated.

File: packages/ts/sso-kit-client-react/package.json
Comment (tamasmak) [inline packages/ts/sso-kit-client-react/package.json:1]: The "main" and the "repository:directory" fields should be updated.

**Comment by manolo** [comment]
Remember to modify TS steps in TC build
https://bender.vaadin.com/admin/editBuildRunners.html?id=buildType:Vaadin6weekProjects_SsoKitAuth_SsoKitRelease
